lang/src/org/partiql/lang/ast/ast.kt

```diff
@@ -26,17 +26,21 @@ sealed class AstNode : Iterable<AstNode> {
     /**
      * returns all the children nodes.
      */
+    @Deprecated("DO NOT USE - will be removed in future revision, see https://github.com/partiql/partiql-lang-kotlin/issues/396")
```

Comment:
Should we specify the appropriate replacement for this?

Reply:
was trying to punt on that, since the discussion of what to use instead is longer than can fit in that comment.  will think of something else.